File: myojana/myojana/report/total_under_process_demands/total_under_process_demands.js
// Copyright (c) 2023, suvaidyam and contributors
// For license information, please see license.txt
var filters = [
	{
		"fieldname": "from_date",
		"fieldtype": "Date",
		"label": "From Date",
	},
	{
		"fieldname": "to_date",
		"fieldtype": "Date",
		"label": "To Date"
	}
	
]
frappe.query_reports["Total under process demands"] = {
	filters: filters,

	"formatter": function(value, row, column, data, default_formatter) {
        // Make the 'name' field clickable to open the relevant document
		console.log(column.fieldname);
        if (column.fieldname === "status" && data) {
            value = `<a href="/app/beneficiary-profiling/?status=${data.status}" style="">${value}</a>`;
        }
        
        value = default_formatter(value, row, column, data);
        return value;
    }
};

File: myojana/myojana/report/total_under_process_demands/total_under_process_demands.py
import frappe
from myojana.utils.report_filter import ReportFilter

def execute(filters=None):
    columns = [
        {
            "fieldname": "status",
            "label": "Current status",
            "fieldtype": "Data",
            "width": 400,
        },
        {
            "fieldname": "count",
            "label": "Count",
            "fieldtype": "Int",
            "width": 200
        }
    ]

    condition_str = ReportFilter.set_report_filters(filters, 'date_of_visit', True, 'ben_table')
    if condition_str:
        condition_str = f"AND {condition_str}"
    else:
        condition_str = ""

    sql_query = f"""
        SELECT
            _sc.status AS status,
            COUNT(_sc.name) AS count
        FROM
            `tabScheme Child` AS _sc
        INNER JOIN `tabBeneficiary Profiling` AS ben_table ON ben_table.name = _sc.parent AND _sc.parenttype = 'Beneficiary Profiling'
        WHERE
            _sc.status = 'Under process' AND
            1=1 {condition_str}
        GROUP BY
            _sc.status;

    """

    data = frappe.db.sql(sql_query, as_dict=True)
    return columns, data
